api/src/main/java/org/pmiops/workbench/workspaces/VwbWorkspaceServiceImpl.java

```diff
@@ -0,0 +1,229 @@
+    workspaceAuthService.validateWorkspaceTierAccess(dbWorkspace);
+
+    RawlsWorkspaceDetails fcWorkspace;
+    WorkspaceResponse workspaceResponse = new WorkspaceResponse();
+
+    WorkspaceDescription workspaceDescription =
+        wsmClient.getWorkspaceAsService(dbWorkspace.getWorkspaceNamespace());
+    fcWorkspace = workspaceMapper.toWorkspaceDetails(workspaceDescription);
+
+    workspaceResponse.setAccessLevel(
+        firecloudMapper.fcToApiWorkspaceAccessLevel(
+            firecloudMapper.fromIamRole(workspaceDescription.getHighestRole())));
+    Workspace workspace =
+        workspaceMapper.toApiWorkspace(dbWorkspace, fcWorkspace, expirationService);
+    workspaceResponse.setWorkspace(workspace);
+
+    return workspaceResponse;
+  }
+
+  @Override
+  public boolean notebookTransferComplete(String workspaceNamespace, String workspaceId) {
+    return false;
+  }
+
+  @Override
+  public List<WorkspaceResponse> getWorkspaces() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public List<WorkspaceResponse> getFeaturedWorkspaces() {
+    logger.warn("getFeaturedWorkspaces not implemented in VWB");
+    return Collections.emptyList();
+  }
+
+  @Override
+  public String getPublishedWorkspacesGroupEmail() {
+    logger.warn("getPublishedWorkspacesGroupEmail not implemented in VWB");
+    return null;
+  }
+
+  @Override
+  public void deleteWorkspace(DbWorkspace dbWorkspace) {
+    logger.warn("deleteWorkspace not implemented for VWB");
+  }
+
+  @Override
+  public void updateWorkspaceBillingAccount(DbWorkspace workspace, String newBillingAccountName) {
+    logger.warn("updateWorkspaceBillingAccount not implemented in VWB");
+  }
+
+  @Override
+  public DbWorkspace saveAndCloneCohortsConceptSetsAndDataSets(DbWorkspace from, DbWorkspace to) {
+    logger.warn("saveAndCloneCohortsConceptSetsAndDataSets not implemented in VWB");
+    return null;
+  }
+
+  @Override
+  public List<UserRole> getFirecloudUserRoles(String workspaceNamespace, String firecloudName) {
+    logger.warn("getFirecloudUserRoles not implemented in VWB");
+    return null;
+  }
+
+  @Override
+  public List<DbUserRecentWorkspace> getRecentWorkspaces() {
+    logger.warn("getRecentWorkspaces not implemented in VWB");
+    return null;
+  }
+
+  @Override
+  public DbUserRecentWorkspace updateRecentWorkspaces(DbWorkspace workspace) {
+    logger.warn("updateRecentWorkspaces not implemented in VWB");
+    return null;
+  }
+
+  @Override
+  public Map<String, DbWorkspace> getWorkspacesByGoogleProject(Set<String> keySet) {
+    logger.warn("getWorkspacesByGoogleProject not implemented in VWB");
+    return null;
+  }
+
+  @Override
+  public DbWorkspace lookupWorkspaceByNamespace(String workspaceNamespace) {
+    logger.warn("lookupWorkspaceByNamespace not implemented in VWB");
+    return null;
+  }
+
+  @Override
+  public void createTanagraStudy(String workspaceNamespace, String workspaceName) {
```

Comment:
I feel like Tanagra code should be moved out to a separate service. Not sure why it's in WorkspaceService...